cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -223,6 +229,120 @@ public BodyConsumer deploy(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Retrieves all adapters in a given namespace.
+   */
+  @GET
+  @Path("/adapters")
+  public void listAdapters(HttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId) {
+    if (!namespaceExists(namespaceId)) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND,
+                           String.format("Namespace '%s' does not exist.", namespaceId));
+      return;
+    }
+    responder.sendJson(HttpResponseStatus.OK, adapterService.getAdapters(namespaceId));
+  }
+
+  /**
+   * Retrieves an adapter
+   */
+  @GET
+  @Path("/adapters/{adapter-name}")
+  public void getAdapter(HttpRequest request, HttpResponder responder,
+                         @PathParam("namespace-id") String namespaceId,
+                         @PathParam("adapter-name") String adapterName) {
+    try {
```

Comment:
Its analogous to the checks on the app methods.
List+create adapters need to check if namespaceExists, whereas the other methods will fail to find the adapterSpec if the namespace does not exist.